var/spack/repos/builtin/packages/ffr/package.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+
+
+class Ffr(MakefilePackage):
+    """FFR: FrontFlow/red, developed in Frontier Simulation Software
+    for Industrial Science(FSIS) project supported by IT program of
+    Ministry of Education, Culture, Sports, Science and Technology(MEXT),
+    JAPAN."""
+
+    homepage = "http://www.ciss.iis.u-tokyo.ac.jp/rss21/theme/multi/fluid/fluid_softwareinfo.html"
+
+    version('3.1.004', sha256='2b396f66bb6437366721fac987f9c6e8b830638c3e4cb5df6a08ff41633f8481', url="file://{0}/FFR_V3.1.004.zip".format(os.getcwd()))
+    version('3.0_000', sha256='edc69fb1fd9dbdb3f531a8f2b9533a9b3c1a28768bb4029b84a6b35c95db0b48', url="file://{0}/open_FrontFlowRed_3.0_000.tar.gz".format(os.getcwd()))
+
+    # FronntFroow/red is used Fortran format I/E without width (Foe Example 3I)
+    # But gfortran require width (For Example (3I6).
+    patch('gfortran_format_31.patch', when='@3.1.004 %gcc')
+    patch('gfortran_format_30.patch', when='@3.0_000 %gcc')
+
+    depends_on('mpi')
+    depends_on('metis@:4', type='link')
+
+    parallel = False
+    build_directories = [
+        join_path('src_main', 'src'),
+        join_path('src_pre', 'src'),
+        join_path('FFR2VIZ', 'src'),
+        join_path('FFR2VIZ', 'src_P_for'),
+    ]
+
+    def edit(self, spec, prefix):
+        flags = ['-O3', '-I.', '-I{0}'.format(spec['metis'].prefix.include)]
+        fflags = flags[:]
+        if spec.satisfies('%gcc'):
+            fflags.append('-ffixed-line-length-none')
+        d = find('.', 'src_main', recursive=True)
+        root_dir  = os.path.dirname(d[0])
+        make = join_path(root_dir, 'src_pre', 'src', 'Makefile')
+        os.chmod(make, 0o644)
+        filter_file('#CSRCS =.*$', 'CSRCS = kmetis_main.c io.c', make)
+        if spec.satisfies('@3.0_000'):
+            d = find('.', 'src_main', recursive=True)
+            root_dir  = os.path.dirname(d[0])
+            for d in ['src_pre', 'FFR2VIZ']:
+                workdir = join_path(root_dir, d, 'src')
+                make = join_path(workdir, 'Makefile')
+                os.chmod(make, 0o644)
+                m = FileFilter(make)
+                m.filter(
+                    r'include Makefile\..*\.in',
+                    'include Makefile.spack.in'
+                )
+                m.filter(
+                    'LIBPRE =.*$',
+                    'LIBPRE = -L{0} -lmetis'.format(spec['metis'].prefix.lib)
```

Comment:
```suggestion
                    'LIBPRE = ' + spec['metis'].libs.ld_flags
```
This has the advantage that it works regardless of whether the libraries are found in `prefix.lib` or `prefix.lib64`.

Reply:
fixed